### PR TITLE
fix: bugfix for claim

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-4040 MF-4113 MF-4114 MF-4115

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugins/Claim/src/SNSAdaptor/components/AirDropActivities/AirDropActivityItem.tsx [hunk 1]
```diff
@@ -161,7 +171,7 @@ export const AirDropActivityItem = memo<AirDropActivityItemProps>(
                                 <Typography className={classes.tips}>{t.no_eligible_tips()}</Typography>
                             ) : null}
                             {isClaimed ? <Typography className={classes.tips}>{t.claimed_tips()}</Typography> : null}
-                            {isEligible && !isClaimed ? (
+                            {true ? (
```
Comment (guanbinrui) [inline packages/plugins/Claim/src/SNSAdaptor/components/AirDropActivities/AirDropActivityItem.tsx]: ?

File: packages/web3-constants/evm/coingecko.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
         "BSCT": "",
         "Matic": "matic-network",
         "Mumbai": "",
-        "Arbitrum": "arbitrum-one",
+        "Arbitrum": "ethereum",
```
Comment (guanbinrui) [inline packages/web3-constants/evm/coingecko.json:37]: Some tokens may exist on arb only.

File: packages/web3-providers/src/Connection/EVM/providers/WalletConnect.ts [hunk 3]
```diff
@@ -217,6 +217,15 @@ export default class WalletConnectProvider
         }
     }
 
+    override async switchChain(chainId?: ChainId | undefined): Promise<void> {
+        return new Promise((resolve, reject) => {
+            super.switchChain(chainId).catch((error) => {
+                reject(error)
+            })
+            this.emitter.on('chainId', () => resolve())
```
Comment (guanbinrui) [inline packages/web3-providers/src/Connection/EVM/providers/WalletConnect.ts]: Listening it in `switchChain`?

File: packages/plugins/Claim/src/hooks/useClaimAirDrop.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import { type SnackbarKey, useCustomSnackbar, type SnackbarMessage, type ShowSnackbarOptions } from '@masknet/theme'
+import { utils } from 'ethers'
+import { useRemoteControlledDialog } from '@masknet/shared-base-ui'
+import { PluginClaimMessage } from '../message.js'
+import { toFixed } from '@masknet/web3-shared-base'
+import { useRef, useCallback } from 'react'
+import { useI18N } from '../locales/i18n_generated.js'
+import { useTheme } from '@mui/material'
+
+export function useClaimAirDrop(
```
Comment (guanbinrui) [inline packages/plugins/Claim/src/hooks/useClaimAirDrop.tsx]:
```suggestion
export function useClaimAirdrop(
```

File: packages/web3-providers/src/Connection/EVM/providers/WalletConnect.ts (hunk 3 above)
Comment (nuanyang233) [reply packages/web3-providers/src/Connection/EVM/providers/WalletConnect.ts]: No, only WalletConnect requires it.

File: packages/web3-constants/evm/coingecko.json (hunk 2 above)
Comment (nuanyang233) [reply packages/web3-constants/evm/coingecko.json:37]: In the Arbitrum network, the arb token does not act as a native token for gas, but rather as an ethereum

File: packages/web3-providers/src/Connection/EVM/providers/WalletConnect.ts (hunk 3 above)
Comment (UncleBill) [inline packages/web3-providers/src/Connection/EVM/providers/WalletConnect.ts]: Don't forget to unsubscribe the event listener

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**